shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala

```diff
@@ -1691,4 +1693,10 @@ object GlutenConfig {
       .doc("Force fallback for orc char type scan.")
       .booleanConf
       .createWithDefault(true)
+
+  val COLUMNAR_NATIVE_CAST_AGGREGATE_ENABLED =
+    buildConf("spark.gluten.sql.columnar.cast.avg")
+      .internal()
+      .booleanConf
+      .createWithDefault(true)
```

Comment:
nit: do we need this config if the cast behavior is default on CH backend?

Reply:
We can disable this type cast when we are sure no overflow will happen, thereby improving the performance of some  cases.

Reply:
@rui-mo agree with this? if yes, merge this pr ?

Reply:
Yes, I understand. Thanks.